src/services/utilServices/__tests__/AuthService.spec.js

```diff
@@ -101,19 +116,76 @@ describe("Auth Service", () => {
     })
   })
 
+  describe("sendOtp", () => {
+    it("should be able to send otp for whitelisted users", async () => {
+      mockUsersService.canSendEmailOtp.mockImplementationOnce(() => true)
+
+      await expect(service.sendOtp(mockEmail)).resolves.not.toThrow()
+      expect(mockUsersService.canSendEmailOtp).toHaveBeenCalledWith(mockEmail)
+      expect(mockUsersService.sendEmailOtp).toHaveBeenCalledWith(mockEmail)
+    })
+
+    it("should throw an error for non-whitelisted users", async () => {
+      mockUsersService.canSendEmailOtp.mockImplementationOnce(() => false)
+
+      await expect(service.sendOtp(mockEmail)).rejects.toThrow(AuthError)
+      expect(mockUsersService.canSendEmailOtp).toHaveBeenCalledWith(mockEmail)
+    })
+  })
+
+  describe("verifyOtp", () => {
+    const mockOtp = "123456"
+    it("should be able to verify otp, login, and return token if correct", async () => {
+      mockUsersService.verifyOtp.mockImplementationOnce(() => true)
+      jwtUtils.signToken.mockImplementationOnce(() => signedToken)
+
+      await expect(
+        service.verifyOtp({ email: mockEmail, otp: mockOtp })
+      ).resolves.toEqual(signedToken)
+      expect(mockUsersService.verifyOtp).toHaveBeenCalledWith(
+        mockEmail,
+        mockOtp
+      )
+      expect(mockUsersService.loginWithEmail).toHaveBeenCalledWith(mockEmail)
+    })
+
+    it("should throw an error if otp is incorrect", async () => {
+      mockUsersService.verifyOtp.mockImplementationOnce(() => false)
+
+      await expect(
+        service.verifyOtp({ email: mockEmail, otp: mockOtp })
+      ).rejects.toThrow(BadRequestError)
+      expect(mockUsersService.verifyOtp).toHaveBeenCalledWith(
+        mockEmail,
+        mockOtp
+      )
+    })
+  })
+
   describe("getUserInfo", () => {
```

Comment:
is it possible for github and email to be both unset and if so, should we add a test case checking that this throws an error? 

should we also check that invalid github ids and emails return errors?

Reply:
It shouldn't be possible for both to be unset - this should be caught by an earlier middleware.

I think the invalid checks belongs in the called methods - otherwise we're just forcing the called method to throw an error